src/notification.c

```diff
@@ -199,19 +199,33 @@ void notification_free(notification * n)
 }
 
 /*
- * Replace all occurrences of "needle" with a quoted "replacement",
- * according to the markup settings.
+ * Replace the two chars where **needle points
+ * with a quoted "replacement", according to the markup settings.
+ *
+ * The needle is a double pointer and gets updated upon return
+ * to point to the first char, which occurs after replacement.
+ *
  */
-char *notification_replace_format(const char *needle, const char *replacement,
-                                  char *haystack, enum markup_mode markup_mode) {
-        char* tmp;
-        char* ret;
+void notification_replace_single_field(char **haystack, char **needle,
+                const char *replacement, enum markup_mode markup_mode) {
 
-        tmp = markup_transform(g_strdup(replacement), markup_mode);
-        ret = string_replace_all(needle, tmp, haystack);
-        g_free(tmp);
+        assert(*needle[0] == '%');
+        // needle has to point into haystack
+        assert(*needle >= *haystack);
+        assert(*needle - *haystack <= strlen(*haystack));
```

Comment:
This assertion should probably test for less than not less-than-or-equal: We don't want a trailing % at the end of the string, if anything it might trip over `string_replace_at` into undefined behaviour.

Reply:
Solved.